src/main.c

```diff
@@ -27,59 +27,95 @@ static int g_sequence = 0;
 
 #define SEQUENCE_MASK (0xffffffff ^ (0xffffffff << SEQUENCE_BITS))
 
-static long get_timestamp() {
+struct conf
+{
+    long snowflake_epoc;
+    int node_id_bits;
+    int datacenter_id_bits;
+    int sequence_bits;
+    int node_id_shift;
+    int datacenter_id_shift;
+    int timestamp_shift;
+    int sequence_mask;
+} conf;
+
+static long get_timestamp()
+{
     struct timeval tv;
     gettimeofday(&tv, NULL);
 
     return tv.tv_sec * 1000 + tv.tv_usec / 1000;
 }
 
-static long get_til_next_millis(long last_timestamp) {
+static long get_til_next_millis(long last_timestamp)
+{
```

Comment:
Do you mind change your indentation setup temporarily so there would be less unnecessary changes